packages/flutter/lib/src/painting/matrix_utils.dart

```diff
@@ -136,23 +196,232 @@ class MatrixUtils {
   /// The transformed rect is then projected back into the plane with z equals
   /// 0.0 before computing its bounding rect.
   static Rect transformRect(Matrix4 transform, Rect rect) {
-    final Offset point1 = transformPoint(transform, rect.topLeft);
-    final Offset point2 = transformPoint(transform, rect.topRight);
-    final Offset point3 = transformPoint(transform, rect.bottomLeft);
-    final Offset point4 = transformPoint(transform, rect.bottomRight);
-    return Rect.fromLTRB(
-        _min4(point1.dx, point2.dx, point3.dx, point4.dx),
-        _min4(point1.dy, point2.dy, point3.dy, point4.dy),
-        _max4(point1.dx, point2.dx, point3.dx, point4.dx),
-        _max4(point1.dy, point2.dy, point3.dy, point4.dy),
-    );
+    final Float64List storage = transform.storage;
+    final double x = rect.left;
+    final double y = rect.top;
+    final double w = rect.right - x;
+    final double h = rect.bottom - y;
+
+    // We want to avoid turning a finite rect into an infinite one if we can.
```

Comment:
We should probably open an issue to discuss whether we can just more safely handle infinity and large/tiny double values more safely - perhaps with a TODO here about removing this if we can do that.

Reply:
I added issue #37277.

Reply:
With respect to removing this protective line of code.  If our own code can have areas where we mishandle an infinite value, would these rectangles potentially fall into code that developers are using that might also mishandle them?

We should definitely harden our own code against mishandling infinite values, and wherever we can we should promote best practices in the apps built on Flutter, but is this test and backup code still useful for protecting the applications outside of our control written against it?

Reply:
It turns out this code will be needed regardless of whether the framework can handle infinite rects because it is returning a rect that has a nan in it. I missed that originally.

When `w` or `h` become infinite and when they are delta transformed by the matrix, we multiply them with 4 matrix elements, some of which can be 0.0. If we multiply an infinity value by 0.0 it becomes nan, so this protective code is needed not just to prevent maxFinite from becoming infinity, but to protect either case from becoming a nan value...